skore/tests/unit/sklearn/test_cross_validate.py

```diff
@@ -35,3 +40,66 @@ def test_cross_validate_return_estimator():
     assert "indices" in reporter.cv_results
     assert "estimator" in reporter._cv_results
     assert "indices" in reporter._cv_results
+
+
+@pytest.mark.parametrize(
+    "estimator,dataset_func,dataset_kwargs,expected_keys",
+    [
+        # Regression case
+        (
+            LinearRegression(),
+            make_regression,
+            {"n_targets": 1},
+            {"r2", "root_mean_squared_error"},
+        ),
+        # Binary classification with predict_proba
+        (
+            LogisticRegression(),
+            make_classification,
+            {"n_classes": 2},
+            {"recall", "precision", "roc_auc", "brier_score_loss"},
+        ),
+        # Binary classification without predict_proba
+        (
+            SVC(probability=False),
+            make_classification,
+            {"n_classes": 2},
+            {"recall", "precision", "roc_auc"},
+        ),
+        # Multiclass classification with predict_proba
+        (
+            LogisticRegression(),
+            make_classification,
+            {"n_classes": 3, "n_clusters_per_class": 1},
+            {
+                "recall_weighted",
+                "precision_weighted",
+                "roc_auc_ovr_weighted",
+                "log_loss",
+            },
+        ),
+        # Multiclass classification without predict_proba
+        (
+            SVC(probability=False),
+            make_classification,
+            {"n_classes": 3, "n_clusters_per_class": 1},
+            {"recall_weighted", "precision_weighted"},
+        ),
+    ],
+    ids=[
```

Comment:
Use [`pytest.param`](https://docs.pytest.org/en/7.1.x/example/parametrize.html#different-options-for-test-ids) to keep the IDs closer to the examples, or create `case_` functions instead (see https://github.com/probabl-ai/skore/blob/29d3689920a77176f92d3dd18ceb52397882b247/skore/tests/unit/sklearn/train_test_split/test_train_test_split.py#L20)